x-pack/plugins/security_solution/public/explore/containers/fields/index.ts

```diff
@@ -5,18 +5,28 @@
  * 2.0.
  */
 
+import { useMemo } from 'react';
+import memoizeOne from 'memoize-one';
 import { useKibana } from '../../../common/lib/kibana';
 
 type FieldValidationCheck = (pattern: string, fieldsList: string[]) => Promise<boolean>;
 
 export const useIsFieldInIndexPattern = (): FieldValidationCheck => {
   const { dataViews } = useKibana().services.data;
-  return async (pattern: string, fieldsList: string[]) => {
-    const fields = await dataViews.getFieldsForWildcard({
-      pattern,
-      fields: fieldsList,
-    });
-    const fieldNames = fields.map((f) => f.name);
-    return fieldsList.every((field) => fieldNames.includes(field));
-  };
+  return useMemo(
+    () =>
+      memoizeOne(
+        async (pattern: string, fieldsList: string[]) => {
```

Comment:
Can we remove the second (`fieldsList`) array parameter and call the `const fieldList = getRequiredMapsFields(pattern);` inside this function? 
This way we simplify the work the _memoizeOne_ has to do. We'll also be able to remove the custom equality function below `(newArgs, lastArgs) => ...`, the default behavior (which is ===) will work.